Plan Chess From Outer Space/Assets/Scripts/BoardManager.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using UnityEngine;
+
+public class BoardManager : MonoBehaviour
+{
+    // --------------- Member variables and data --------------- //
+    // ChessBoard Settings
+    [Header("Chess Board Settings")] 
+    [SerializeField, Tooltip("Chess board width.")]
+    private int _boardWidth = 8;
+    [SerializeField, Tooltip("Chess board depth.")]
+    private int _boardDepth = 8;
+    [SerializeField, Tooltip("Chess board center position.")]
+    private Vector3 _centerPosition;
+    // This list contains every board square generated on initialization
+    private List<BoardSquare> _boardSquares;
+
+    // BoardSquare Settings
+    [Header("Board Squares")]
+    [SerializeField, Tooltip("Prefab that is used for the board squares.")] 
+    private GameObject _squareBoardPrefab;
+    [SerializeField, Tooltip("Materials for the board squares")]
+    private Material[] _materialsArray;
+    private int _materialCount;
+    
+    // Chess pieces information
+    [Header("Chess Pieces Information and Properties")] 
+    [SerializeField, Tooltip("Pawn Chess Piece")]
+    private GameObject _pawnPrefab;
```

Comment:
We can index the **ChessPieceType** later. As you said in the comment above, we need to change the enum to another class, so indexing that now, won't be useful because the enum is going to be changed soon.